content/en/blog/_posts/2022-11-11-retroactive-default-storage-class.md

```diff
@@ -0,0 +1,131 @@
+
+Looking at the scenarios above it is clear that neither of them is optimal.
+
+### Resource ordering during cluster installation
+
+If cluster installation tool needs to create resources that require storage (e.g. image registry) it was difficult to get the ordering right because any Pods that require storage and rely on default StorageClass presence will fail to create if there is no default at that point yet.
+
+## What changed
+
+We've changed PersistentVolume (PV) controller in KCM, so it can assign default StorageClass to unbound PersistentVolumeClaims that have storageClassName set to `nil`. In order to allow the value change PersistentVolumeClaim admission in the API server also changed to allow changing the value from `nil` to actual StorageClass name.
+
+### `nil` vs `""` for storageClassName - does it matter?
+
+Before this feature was introduced those values were equal in terms of behavior and any PersistentVolumeClaim with `nil` or `""` storageClassName value would bind to existing PersistentVolume resource with storageClassName also set to `nil` or `""`.
+
+With this new feature enabled we wanted to maintain this behavior but also be able to update the StorageClass name. With these constraints in mind the feature changes the semantics of `nil` if default StorageClass is present to always mean "Give me a default" and `""` to mean "Give me PersistentVolume that also has `""` StorageClass name".
+
+In other words we need to distinguish two main cases, either default StorageClass is not present in which case there's no behavior change or where there is a default in which case PVCs with `nil` won't bind to PVs anymore and instead will be updated to have the StorageClass.
+
+The tables below show all these cases to better describe when PVC binds and when it's StorageClass gets updated.
+
+| Without default class        | PVC storageClassName == `""` | PVC storageClassName == `nil` |
+|------------------------------|------------------------------|-------------------------------|
+| PV storageClassName == `""`  | binds                        | binds                         |
+| PV without storageClassName  | binds                        | binds                         |
+
+
+| With default class          | PVC storageClassName == `""` | PVC storageClassName == `nil` |
+|-----------------------------|------------------------------|-------------------------------|
+| PV storageClassName == `""` | binds                        | class updates                 |
+| PV without storageClassName | binds                        | class updates                 |
+
+## How to use it
+
+You don't have to perform any configuration steps - the feature will work "out of the box" once it's GA.
```

Comment:
Is the feature GA? If not, don't write "You don't have to perform any configuration steps" in present tense.

Reply:
I wanted to make it clear that in Beta and GA it's already enabled and no configuration is needed. Maybe it's ok to leave just the sentence below that alpha testing needs feature gate and remove this one - the rest is understood.

Reply:
Maybe